[AustinHuang1203] iP

AustinHuang1203 opened the pull request:
#Termina
- A good bot
- Helps to store your tasks
- Is polite
1. Can take in todos, events, etc
2. Can delete items 
3. Can mark items done
```python
while true == true:
    print("Hello bot")
```
- [x] Tested
- [x] Works perfectly
- [ ] Free to use
🤡 
> This is a blockquote.
> Please use my bot
> Its only 10 dollars to use

[Some link](https://github.com)

So if you are confused, just run `git help` or something
```
java run start
git blame
```
So I **hope** you *use* my ~~overpowered~~ bot :)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

AustinHuang1203 committed:
First commit

AustinHuang1203 committed:
first

AustinHuang1203 committed:
level2

AustinHuang1203 committed:
part 3 done

AustinHuang1203 committed:
level 4

AustinHuang1203 committed:
testing

AustinHuang1203 committed:
error handling

AustinHuang1203 committed:
added delete

AustinHuang1203 committed:
some edits

AustinHuang1203 committed:
Implement Level-7

Current state: Does not save data to hard disk and just saves it in the program memory

Why problem:Not good, does not meet task requirements and we cant save it after our program shuts down

This commit: Saves the tasklist to harddrive and loads it when you open ur program

AustinHuang1203 committed:
Merge pull request #1 from AustinHuang1203/branch-Level-7

Implement Level-7

AustinHuang1203 committed:
Implement Level-8

Current state: No date time formatter, just accepts string

Why problem:Not good as can key in anything

This commit: Fixes this by formatting date for event and deadline

This commit: Saves the tasklist to harddrive and loads it when you open ur program

AustinHuang1203 committed:
Merge pull request #2 from AustinHuang1203/branch-Level-8

Implement Level-8

AustinHuang1203 committed:
Implement A-MoreOOP

Current state: Some OOP but can be better

Why problem:Not OOP enough!

This commit: Makes it more OOP by creating 4 new classes

AustinHuang1203 committed:
Merge pull request #3 from AustinHuang1203/branch-Level-8

Implement A-MoreOOP

AustinHuang1203 committed:
Implement A-Packages

Current state: Classes not in package

Why problem: Not nice

This commit: Added classes to package

AustinHuang1203 committed:
Bug fix

Current state: the test file not updated

Why problem: test dosent run

This commit: fixed filepath

AustinHuang1203 committed:
Merge pull request #4 from AustinHuang1203/add-gradle-support

Add gradle support

AustinHuang1203 committed:
Implement A-Gradle
Current: Does not use gradle
Why problem: Gradle is useful tool to use
this fix: I added gradle and configured the run task in gradle

AustinHuang1203 committed:
Implement A-JUnit
Current: No test cases
Why problem: We need test cases
this fix: Added test cases for 2 classes

AustinHuang1203 committed:
Implement javadoc
Current: Few comments
Why problem: We need documentation to understand code
this fix: Added documentation

AustinHuang1203 committed:
Implement javastyle
Current: Ugly code
Why problem: Not nice to read
this fix: Make code look nicer and more readable
Why this way: Improve efficiency and style

AustinHuang1203 committed:
Implement Level-9
Current: No find function
Why problem: Cant find item in list
this fix: Added find function
Why this way: Can find stuff

AustinHuang1203 committed:
Merge pull request #5 from AustinHuang1203/branch-A-JavaDoc

Implement javadoc

AustinHuang1203 committed:
Merge branch 'master' into branch-A-CodingStandard

AustinHuang1203 committed:
Merge pull request #6 from AustinHuang1203/branch-A-CodingStandard

Implement javastyle

AustinHuang1203 committed:
Merge branch 'master' into branch-Level-9

AustinHuang1203 committed:
Merge pull request #7 from AustinHuang1203/branch-Level-9

Implement Level-9

AustinHuang1203 committed:
Implement Level-10
Current: No GUI
Why problem: Hard to use and ugly as need to read from the terminal output
this fix: Made a gui using javafx for the program so easier to read
Why this way: Easy to use interface that is nice to look at

AustinHuang1203 committed:
Merge pull request #8 from AustinHuang1203/branch-Level-10

Implement Level-10

AustinHuang1203 committed:
Implement A-Checkstyle
Current: No checkstyle ran
Why problem: Some code violations I might have missed out on
this fix: Ran the autochecker so I could resolve all the reamining issues
Why this way: Easier to use than manually going through the code

jannnice commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,36 @@
+ *
+ */
+public class Task implements Serializable {
+    public Boolean done;
```
for this variable, since it is a boolean, I think it's better to name it to sound more like a boolean. for example : isDone

AustinHuang1203 replied:
Thanks for the feedback!

jannnice commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,69 @@
+    private final String filepath;
+
+    /**
+     * constructor for filepath
```
The content of the javadoc is nice and clear! What can be improved would be the use of proper capitalisation and punctuation (period to end the sentence).

jannnice reviewed (commented):
Overall, everything looks nicely structured. The javadoc and comments really helped in understanding what each function does.

InfiBeyond commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class DialogBox extends HBox {
+
+    private Label text;
```
Constant names should be in all caps.

AustinHuang1203 replied:
Thanks for the feedback!

InfiBeyond commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,162 @@
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(getResponse(userInput.getText()));
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, new ImageView(user)),
+                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+        );
```
Great use of line break!
The ending parenthesis can stay with the previous line.
i.e.
dialogContainer.getChildren().addAll(
                DialogBox.getUserDialog(userText, new ImageView(user)),
                DialogBox.getDukeDialog(dukeText, new ImageView(duke)) );

InfiBeyond commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,46 @@
+    public String display() {
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        if (done) {
+            return "[E][X] " + this.name + " (From: " +
+                    start.format(outputFormatter) + " To: " + end.format(outputFormatter) + ")";
```
Place line break before an operator.
i.e. 
return "[E][X] " + this.name + " (From: " 
                    + start.format(outputFormatter) + " To: " + end.format(outputFormatter) + ")";

InfiBeyond commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,46 @@
+            return "[E][X] " + this.name + " (From: " +
+                    start.format(outputFormatter) + " To: " + end.format(outputFormatter) + ")";
+        }
+        return "[E][] " + this.name + " (From: " +
+                start.format(outputFormatter) + " To: " + end.format(outputFormatter) + ")";
```
Same as above, place line break before an operator.

InfiBeyond commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+    public TaskList parse(String stuff, TaskList items) {
+
+        if (stuff.equalsIgnoreCase("list")) {
+
```
unnecessary line break

InfiBeyond commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+            for (int i = 0; i < items.size(); i++) {
+                update((i + 1) + ". " + items.get(i).display());
+            }
+
```
unnecessary line break

InfiBeyond commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,60 @@
+     * gives size
+     * @return size
+     */
+    public int size() {
```
Names representing methods must be verbs. 
i.e. getSize()

InfiBeyond commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,49 @@
+/**
+ * The UI class to handle user interactions
+ */
+public class UI {
```
Abbreviations and acronyms should not be uppercase when used as a part of a name. We can change the name to Ui.

InfiBeyond reviewed (commented):
Overall, the code has been great following the Java coding standard. Writing the JavaDoc description section in complete sentences and proper capitalization would be better.

AlainS87 commented on src/main/java/duke/Deadline.java:
Great to see you have followed the coding stardard, and added concrete descriptions in javadoc, a small advice would be to change duke to other name, so that your chatbot can have more features belong to your own style.

AlainS87 commented on src/main/java/duke/TaskList.java:
Seems like you are not following coding standard when writing this size() method, you can try to change its name to getSize()

AlainS87 commented on src/main/resources/images/DaDuke.png:
change the picture so the bot can be more creative

AlainS87 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+        }
+        return "[D][] " + this.name + " (Due By: " + by.format(outputFormatter) + ")";
+    }
+}
```
Great to see you have followed the coding stardard, and added concrete descriptions in javadoc, a small advice would be to change duke to other name, so that your chatbot can have more features belong to your own style.

AlainS87 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,60 @@
+    public ArrayList<Task> toArray() {
+        return items;
+    }
+}
```
Seems like you are not following coding standard when writing this size() method, you can try to change its name to getSize()

AustinHuang1203 committed:
Implement A-Checkstyle
Current: No checkstyle ran
Why problem: Some code violations I might have missed out on
this fix: Ran the autochecker so I could resolve all the reamining issues
Why this way: Easier to use than manually going through the code

AustinHuang1203 committed:
Implement A-Assertions
Current: No assertions
Why problem: Without assertions, we cannot check if our code is maintaining some assumptions which could lead to dangerous bugs!
this fix: Added some assertions to check assumptions at different points in code
Why this way: Make sure we know if any assumptions are not being held at any point in the code as they could cause bugs

AustinHuang1203 committed:
Implement A-CodeQuality
Current: Good quality code but with some minor issues
Why problem: There are minor lapses in code quality which can impact readability
this fix: Made some minor improvements to formatting and code quality like variable name change
Why this way: Complies to the given codequality standards which standardises code and makes it more readable to work on

AustinHuang1203 committed:
Merge pull request #10 from AustinHuang1203/brach-A-Assertions

Implement A-Assertions

AustinHuang1203 committed:
Merge branch 'master' into branch-A-CodeQuality

AustinHuang1203 committed:
Merge pull request #9 from AustinHuang1203/branch-A-CodeQuality

Implement A-CodeQuality

AustinHuang1203 committed:
Implement BCD-Extension (C-Help)
Current: No helppage!!!
Why problem: The user cannot seek help if they need to
this fix: Added a help command to display a simple help page
Why this way: So the user knows what commands are available when calling help

AustinHuang1203 committed:
Implement A-UserGuide
Current: No readme
Why problem: There is no good readme instructions
this fix: Updated readme
Why this way: So the user has an instruction manual on how to use

AustinHuang1203 committed:
Implement A-UserGuide
Current: No readme
Why problem: There is no good readme instructions
this fix: Updated readme
Why this way: So the user has an instruction manual on how to use

AustinHuang1203 committed:
Implement A-UserGuide
Current: No readme
Why problem: There is no good readme instructions
this fix: Updated readme
Why this way: So the user has an instruction manual on how to use
